src/Core/Geometry/Volume.hpp

```diff
@@ -186,6 +186,10 @@ class RA_CORE_API VolumeGrid : public AbstractDiscreteVolume
 
     using Container = std::vector<ValueType>;
 
+    // For now, assume that AbstractDiscreteVolume::ValueType is Scalar
+    // vector4 : gradient and density
```

Comment:
```suggestion
    // Since AbstractDiscreteVolume::ValueType is scalar (see class documentation),
    // hence gradient is a Vector3<ValueType>. 
    // GradientContainer also store density as forth component
```